### PR TITLE
Separate process packet into subfunctions

### PR DESCRIPTION
**Pull request by Brett208**
This branch is mostly to improve readability and future maintainability by reducing a function size

### CODE REVIEW

File: GameServer.cpp [hunk 1]
```diff
@@ -172,9 +172,6 @@ int GameServer::AllocSocket(SOCKET& hostSocket, unsigned short port)
 
 void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
 {
-	unsigned int gameInfoIndex;
```
Comment (Brett208) [inline GameServer.cpp:173]: When splitting the function, I noticed gameInfoIndex is defined as unsigned, but is being checked against a value of -1. I think the unsigned qualifier needs to be dropped. However, it becomes a large changeset, so would like to solve in another branch to keep this one from ballooning in proportion.

File: GameServer.cpp [hunk 2]
```diff
@@ -183,182 +180,206 @@ void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
...
-		gameInfoIndex = FindGameInfoClient(from, packet.tlMessage.gameServerPoke.randValue);
+	// Find the current GameInfo
+	unsigned int gameInfoIndex = FindGameInfoClient(from, packet.tlMessage.gameServerPoke.randValue);
 
-		// Check what kind of poke this is
-		switch(packet.tlMessage.gameServerPoke.statusCode)
+	// Check what kind of poke this is
+	switch (packet.tlMessage.gameServerPoke.statusCode)
+	{
+	case pscGameHosted: {
```
Comment (Brett208) [inline GameServer.cpp:297]: cases within this switch need to be bracketed due to defining 'GameInfo* currentGameInfo' within the switch statement now

File: GameServer.cpp (hunk 1 above)
Comment (DanRStevens) [reply GameServer.cpp:173]: It's an index value, so `std::size_t` is probably most appropriate. The `-1` sentinel value could be replaced by `numeric_limits<T>::max()`.

Assuming it's never serialized into a fixed size field in a packet anyway. In that case, a `std::uint32_t` would probably be better.

File: GameServer.cpp [hunk 3]
```diff
@@ -183,181 +180,205 @@ void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
...
+	// Check if they have a different internal address
+	if (from.sin_port != internalPort)
+	{
+		// Send second reply
+		packet.tlMessage.echoExternalAddress.replyPort = internalPort;
+		from.sin_port = htons(internalPort);
+		SendTo(packet, from);
+	}
+
+
```
Comment (DanRStevens) [inline GameServer.cpp]: Extra blank lines.

File: GameServer.cpp [hunk 4]
```diff
@@ -183,181 +180,205 @@ void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
...
+	case pscGameCancelled: {
+		LogEndpoint("Game Cancelled: ", from.sin_addr.s_addr, from.sin_port);
 
-		break;
+		// Remove the game from the list
+		FreeGameInfo(gameInfoIndex);
+		// Update counters
+		counters.numGamesCancelled++;
+
+		break; }
```
Comment (DanRStevens) [inline GameServer.cpp]: Could probably put the close brace on it's own line.

File: GameServer.cpp [hunk 5]
```diff
@@ -183,181 +180,205 @@ void GameServer::ProcessPacket(Packet &packet, sockaddr_in &from)
...
 
-		// Check if they have a different internal address
-		if (from.sin_port != internalPort)
-		{
-			// Send second reply
-			packet.tlMessage.echoExternalAddress.replyPort = internalPort;
-			from.sin_port = htons(internalPort);
-			SendTo(packet, from);
-		}
+		break; }
```
Comment (DanRStevens) [inline GameServer.cpp]: Closing brace.

**Review by DanRStevens — APPROVED** [review]
Looks good.

Noted some minor optional style adjustments.

**Review by DanRStevens — APPROVED** [review]

**Pull request merged**